scanner/matcher/service.go

```diff
@@ -35,6 +36,12 @@ func (s *matcherService) RegisterServiceServer(grpcServer *grpc.Server) {
 	v4.RegisterMatcherServer(grpcServer, s)
 }
 
+// AuthFuncOverride specifies the auth criteria for this API.
+func (s *matcherService) AuthFuncOverride(ctx context.Context, fullMethodName string) (context.Context, error) {
+	// TODO: Setup permissions for matcher.
+	return ctx, allow.Anonymous().Authorized(ctx, fullMethodName)
```

Comment:
(repeat) Recommend adding a failing test for anonymous access just in case this TODO is missed, that way an unauthenticated API is less likely to be released.

Reply:
https://github.com/stackrox/stackrox/pull/6663#discussion_r1267004040